lightblue/src/test/java/org/esbtools/eventhandler/lightblue/LightblueNotificationRepositoryTest.java

```diff
@@ -238,8 +239,9 @@ public void shouldRetrieveNonOverlappingSetsOfNotificationsIfCalledByMultipleThr
             bothThreadsStarted.await();
 
             Thread.sleep(5000);
-
             thread2Client.unpause();
+            
+            Thread.sleep(5000);
```

Comment:
We don't want to sleep again here I think. In fact, ideally we would be able to unpause both clients at *exactly* the same time for maximal parallelism.

That said, I did have to make an update to SlowDataLightblueClient for this test to work. At line 140 there is currently:

```java
        } catch (Exception e) {
            if (e instanceof RuntimeException) {
                throw (RuntimeException) e;
            }

            throw new RuntimeException(e);
        }
```

I had to make this:

```java
        } catch (Exception e) {
            if (e instanceof LightblueException) {
                throw (LightblueException) e;
            }

            if (e instanceof RuntimeException) {
                throw (RuntimeException) e;
            }

            throw new RuntimeException(e);
        }
```